### PR TITLE
`ultralytics 8.3.2` fix AMP checks with `imgsz=256`

### LINKED ISSUE #16582: AMP Checks failing since 8.3.1
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

I upgraded to 8.3.1 yesterday and wanted to retrain my segmentation/classification/object detection models with the new Yolo11 model architecture.
However since 8.3.1 it disables AMP during training startup due to AMP checks failing, it did not do this before?
I have a hunch that this is due to the CPU not supporting AVX2 which causes issues with NNPACK

```
python train.py
[W1001 08:42:50.413718206 NNPACK.cpp:61] Could not initialize NNPACK! Reason: Unsupported hardware.
YOLOv8s-cls summary: 99 layers, 6,361,736 parameters, 6,361,736 gradients, 13.6 GFLOPs
Ultralytics 8.3.1 \U0001f680 Python-3.12.3 torch-2.4.1+cu124 CUDA:0 (NVIDIA GeForce RTX 3060, 11931MiB)
engine/trainer: task=classify, mode=train, model=yolov8s-cls.yaml, data=dataset/images, epochs=300, time=None, patience=25, batch=16, imgsz=640, save=True, save_period=25, cache=disk, device=0, workers=8, project=None, name=classification-s, exist_ok=True, pretrained=False, optimizer=AdamW, verbose=True, seed=0, deterministic=False, single_cls=False, rect=False, cos_lr=True, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.1, val=True, split=val, save_json=False, save_hybrid=False, conf=0.5, iou=0.5, max_det=100, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=True, opset=None, workspace=4, nms=False, lr0=5e-05, lrf=0.01, momentum=0.937, weight_decay=0.0002, warmup_epochs=10, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.3, hsv_v=0.4, degrees=0.2, translate=0.1, scale=0.5, shear=0.2, perspective=0.0002, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.0, copy_paste_mode=flip, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=runs/classify/classification-s

train: /var/share/classification/dataset/images/train... found 449 images in 107 classes \u2705 
val: /var/share/classification/dataset/images/val... found 144 images in 107 classes \u2705 
test: None...
Overriding model.yaml nc=1000 with nc=107

                   from  n    params  module                                       arguments                     
  0                  -1  1       928  ultralytics.nn.modules.conv.Conv             [3, 32, 3, 2]                 
  1                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]                
  2                  -1  1     29056  ultralytics.nn.modules.block.C2f             [64, 64, 1, True]             
  3                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               
  4                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]           
  5                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]              
  6                  -1  2    788480  ultralytics.nn.modules.block.C2f             [256, 256, 2, True]           
  7                  -1  1   1180672  ultralytics.nn.modules.conv.Conv             [256, 512, 3, 2]              
  8                  -1  1   1838080  ultralytics.nn.modules.block.C2f             [512, 512, 1, True]           
  9                  -1  1    794987  ultralytics.nn.modules.head.Classify         [512, 107]                    
YOLOv8s-cls summary: 99 layers, 5,217,803 parameters, 5,217,803 gradients, 12.7 GFLOPs

AMP: running Automatic Mixed Precision (AMP) checks with YOLO11n...
AMP: checks failed \u274c. Anomalies were detected with AMP on your system that may lead to NaN losses or zero-mAP results, so AMP will be disabled during training.
train: Scanning /var/share/classification/dataset/images/train... 449 images, 0 corrupt: 100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 449/449 [00:00<?, ?it/s]
val: Scanning /var/share/lassification/dataset/images/val... 144 images, 0 corrupt: 100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 144/144 [00:00<?, ?it/s]
optimizer: AdamW(lr=5e-05, momentum=0.937) with parameter groups 26 weight(decay=0.0), 27 weight(decay=0.0002), 27 bias(decay=0.0)

Image sizes 640 train, 640 val
Using 2 dataloader workers
Logging results to runs/classify/classification-s
Starting training for 300 epochs...

      Epoch    GPU_mem       loss  Instances       Size
      1/300      5.02G       4.77         16        640:  17%|        | 5/29 [00:03<00:18,  1.27it/s]
```

I did order a i5-7100t yesterday to replace the CPU in my training rig which will be arriving in a couple of days, but I just want to confirm that is the actual cause of this issue and not something else.
Before 8.3.1 my AMP checks did not fail and I did successfully train models so not sure why it's an issue now all of a sudden.

### Environment

Ultralytics 8.3.1 🚀 Python-3.12.3 torch-2.4.1+cu124 CUDA:0 (NVIDIA GeForce RTX 3060, 11931MiB)
Setup complete ✅ (2 CPUs, 31.3 GB RAM, 132.3/230.7 GB disk)

OS                  Linux-6.8.0-45-generic-x86_64-with-glibc2.39
Environment         Linux
Python              3.12.3
Install             pip
RAM                 31.27 GB
CPU                 Intel Pentium G4560 3.50GHz
CUDA                12.4

numpy               ✅ 1.26.3<2.0.0,>=1.23.0
matplotlib          ✅ 3.9.2>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.2.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.14.1>=1.4.1
torch               ✅ 2.4.1+cu124>=1.8.0
torchvision         ✅ 0.19.1+cu124>=0.9.0
tqdm                ✅ 4.66.5>=4.64.0
psutil              ✅ 6.0.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.2>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.5>=2.0.0
torch               ✅ 2.4.1+cu124!=2.4.0,>=1.8.0; sys_platform == "win32"

### Minimal Reproducible Example

training_defaults.py
```
epochs = 300
patience = 25
warmup_epochs = 10
hsv_s = 0.3
weight_decay = 0.0002
iou = 0.5
max_det = 100
conf = 0.5
learning_rate = 0.00005
dropout = 0.1
perspective = 0.0002
shear=0.2
degrees=0.2
batch_size=16
nominal_batch_size=64
```

train.py
```
import comet_ml
from ultralytics import YOLO
import training_defaults
import os
import argparse
import torch

comet_ml.login(project_name="classification")

if __name__ == '__main__':
    # Create an ArgumentParser object
    parser = argparse.ArgumentParser(description="Classification trainer")
    
    # Add arguments
    parser.add_argument('-n', '--name', type=str, default="classification", help='Training run name', required=False)
    parser.add_argument('-c', '--cache', type=str, default="ram", help='Cache type', required=False)
    parser.add_argument('-o', '--optimizer', type=str, default="AdamW", help='Optimizer to use', required=False)
    parser.add_argument('-s', '--scale', type=str, default="s", help='Model scale (n,s,m,l,x)', required=False)
    parser.add_argument('-d', '--deterministic', type=bool, default=False, help='Forces deterministic algorithm use', required=False)
    parser.add_argument('-r', '--resume', type=bool, default=False, help='Resume from last checkpoint', required=False)
    parser.add_argument('-e', '--epochs', type=int, default=training_defaults.epochs, help='Total epochs to train', required=False)
    parser.add_argument('-p', '--patience', type=int, default=training_defaults.patience, help='Total epochs to train', required=False)
    parser.add_argument('-we', '--warmup_epochs', type=int, default=training_defaults.warmup_epochs, help='Number of epochs for learning rate warmup', required=False)
    parser.add_argument('-is', '--image_size', type=int, default=640, help='Target image size for training', required=False)
    parser.add_argument('-b', '--batch', type=int, default=training_defaults.batch_size, help='Batch size to train with', required=False)
    parser.add_argument('-lr', '--learning_rate', type=float, default=training_defaults.learning_rate, help='Initial learning rate', required=False)
    parser.add_argument('-sp', '--save_period', type=int, default=25, help='Frequency of saving model checkpoints', required=False)
    parser.add_argument('-mg', '--multi_gpu', type=bool, default=False, help='Use all available GPU\'s', required=False)
    
    # Parse the arguments
    args = parser.parse_args()

    training_name = f"{args.name}-{args.scale}"
    training_folder = f"runs/classify/{training_name}"
    
    # load pretrained model
    model_path = f"yolo11{args.scale}-cls.yaml"
    
    if args.resume == True:
        # resume model from last checkpoint
        model_path = f'./{training_folder}/weights/last.pt'
    
    # Load rtdetr model or resume from checkpoint
    model = YOLO(model_path)
    
    device = 0
    
    if args.multi_gpu:
        device = list([i for i in range(torch.cuda.device_count())])

    # Train the model
    results = model.train(
            data="dataset/images",
            epochs=args.epochs,
            warmup_epochs=args.warmup_epochs,
            imgsz=args.image_size,
            patience=args.patience,
            deterministic=args.deterministic,
            save_period=args.save_period,
            amp=True,
            #cache=args.cache,
            cache="disk",
            batch=args.batch,
            name=training_name,
            resume=args.resume,
            lr0=args.learning_rate,
            pretrained=False,
            weight_decay=training_defaults.weight_decay,
            iou=training_defaults.iou,
            max_det=training_defaults.max_det,
            conf=training_defaults.conf,
            optimizer=args.optimizer,
            device=device,
            exist_ok=True,
            cos_lr=True,
            dropout=training_defaults.dropout,
            plots=True,
            #augmentation settings
            hsv_s=training_defaults.hsv_s,
            perspective=training_defaults.perspective,
            shear=training_defaults.shear,
            degrees=training_defaults.degrees
    )

```

### Additional

PS: for whatever reason yolo checks reports I am using a G4560, I swapped it out to a G3900 to attempt to fix the issue to no avail, lscpu correctly reports using a G3900 but for some reason yolo checks still reports the old cpu.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

### LINKED ISSUE #16585: Problems with AMP - yolov11n - Google Colab
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

```rt-detr-l summary: 681 layers, 32,808,131 parameters, 32,808,131 gradients, 108.0 GFLOPs

Transferred 926/941 items from pretrained weights
TensorBoard: Start with 'tensorboard --logdir runs/detect/fullImage_arrowShape-v12', view at http://localhost:6006/
AMP: running Automatic Mixed Precision (AMP) checks with YOLO11n...
AMP: checks failed ❌. Anomalies were detected with AMP on your system that may lead to NaN losses or zero-mAP results, so AMP will be disabled during training.
train: Scanning /content/FullImage-ArrowANDShape-2/train/labels.cache... 26 images, 0 backgrounds, 0 corrupt: 100% 26/26 [00:00<?, ?it/s]
/usr/local/lib/python3.10/dist-packages/albumentations/__init__.py:13: UserWarning: A new version of Albumentations is available: 1.4.17 (you have 1.4.15). Upgrade using: pip install -U albumentations. To disable automatic update checks, set the environment variable NO_ALBUMENTATIONS_UPDATE to 1.
  check_for_updates()
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, num_output_channels=3, method='weighted_average'), CLAHE(p=0.01, clip_limit=(1, 4.0), tile_grid_size=(8, 8))
val: Scanning /content/FullImage-ArrowANDShape-2/valid/labels.cache... 3 images, 0 backgrounds, 0 corrupt: 100% 3/3 [00:00<?, ?it/s]
Plotting labels to runs/detect/fullImage_arrowShape-v12/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.002, momentum=0.9) with parameter groups 143 weight(decay=0.0), 206 weight(decay=0.0005), 226 bias(decay=0.0)
```

### Environment

```
Ultralytics 8.3.1 🚀 Python-3.10.12 torch-2.4.1+cu121 CUDA:0 (NVIDIA L4, 22700MiB)
Setup complete ✅ (12 CPUs, 53.0 GB RAM, 36.6/112.6 GB disk)

OS                  Linux-6.1.85+-x86_64-with-glibc2.35
Environment         Colab
Python              3.10.12
Install             pip
RAM                 52.96 GB
CPU                 Intel Xeon 2.20GHz
CUDA                12.1

numpy               ✅ 1.26.4<2.0.0,>=1.23.0
matplotlib          ✅ 3.7.1>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.4.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.13.1>=1.4.1
torch               ✅ 2.4.1+cu121>=1.8.0
torchvision         ✅ 0.19.1+cu121>=0.9.0
tqdm                ✅ 4.66.5>=4.64.0
psutil              ✅ 5.9.5
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.1.4>=1.1.4
seaborn             ✅ 0.13.1>=0.11.0
ultralytics-thop    ✅ 2.0.8>=2.0.0
torch               ✅ 2.4.1+cu121!=2.4.0,>=1.8.0; sys_platform == "win32"
```

### Minimal Reproducible Example

```
!yolo train model=rtdetr-l.pt data=/content/FullImage-ArrowANDShape-2/data.yaml imgsz=1024 epochs=100 patience=20 batch=1 name=fullImage_arrowShape-v1
```

### Additional

_No response_

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by glenn-jocher**
https://github.com/ultralytics/ultralytics/issues/16582

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Upgrade to version 8.3.2 includes improvements for CUDA operations and deterministic training settings.

### 📊 Key Changes
- Added a new test to check AMP (Automatic Mixed Precision) compatibility.
- Changed the image size in AMP compatibility checks for better accuracy.
- Updated environment variable settings to ensure deterministic training.
- Version bump from 8.3.1 to 8.3.2.

### 🎯 Purpose & Impact
- **AMP Testing**: Ensures models are compatible with faster mixed precision training, optimizing speed without losing accuracy. 🚀
- **Deterministic Training**: Reduces variance in results, helping in achieving consistent model performance across runs. 🔄
- **Version Update**: Represents improved stability and functionality in the software. ⚙️

### CODE REVIEW

**Comment by glenn-jocher** [comment]
@ambitious-octopus this warning is now showing after you removed the ENV variables a few days ago. Can you revert this change please?

Screenshot 2024-10-01 at 10 24 27
